lang/shared/src/main/scala/com/wavesplatform/lang/v1/parser/Parser.scala

```diff
@@ -249,17 +249,19 @@ object Parser {
   def matchCaseP(implicit c: fastparse.P[Any]): P[MATCH_CASE] = {
     def checkForGenericAndGetLastPos(t: Type): Either[INVALID, Option[Pos]] =
       t match {
+        case Single(VALID(position, "List"), Some(VALID(_, AnyType(_)))) => Right(Some(position))
         case Single(name, parameter) =>
           parameter
             .toLeft(Some(name.position))
             .leftMap {
-              case VALID(position, v)              => INVALID(position, s"Unexpected generic match type [$v]")
+              case VALID(position, v)              => INVALID(position, s"Unexpected generic match type $t")
```

Comment:
Why not the `[$v]` but more common `$t`?

Reply:
For ```case x:List[Int] => ...``` old code produced error "Unexpected generic match type [Int]", new code produce more informative "Unexpected generic match type List[Int]".